### PR TITLE
add app.register_listener method

### PR DESCRIPTION
**Pull request by knowsuchagency**
Simply adds a `register_listener` method to the Sanic app, similar to the [register_middleware](https://sanic.readthedocs.io/en/latest/sanic/api_reference.html#sanic.app.Sanic.register_middleware) method

### CODE REVIEW

File: sanic/app.py [hunk 1]
```diff
@@ -117,6 +117,21 @@ def decorator(listener):
 
         return decorator
 
+    def register_listener(self, listener, event):
+        """
+        Register the listener for a given event.
+
```
Comment (r0fls) [inline sanic/app.py]: can you use single spacing here in the doc string?

Comment (knowsuchagency) [reply sanic/app.py]: Removed erroneous whitespace in a new commit

**Comment by r0fls** [comment]
@knowsuchagency last thing: can you add a small piece to the docs about using this? It would go in here: docs/sanic/middleware.md

**Comment by knowsuchagency** [comment]
Done!

**Pull request merged**